src/storage-operators/src/s3_oneshot_sink.rs

```diff
@@ -391,6 +406,14 @@ where
                                     uploader.append_row(&row).await?;
                                 }
                             }
+                            // A very crude way to detect if there is ever a regression in the deterministic
+                            // ordering of rows in our input, since we are depending on an implementation
+                            // detail of timely communication (FIFO ordering over an exchange).
+                            let cur = (row, batch);
+                            if let Some(last) = last_row {
+                                assert!(&last < &cur, "broken fifo ordering!");
```

Comment:
wait, what if it `last == cur`?

Reply:
I believe since we consolidated all inputs that any duplicated rows would just have a `diff > 1`, right?

Reply:
Yes, that's correct!